zerver/tests/test_users.py

```diff
@@ -1950,6 +1951,270 @@ def test_email_sent(self) -> None:
         self.assertEqual(msg.reply_to[0], "noreply@testserver")
         self.assertIn("Dear Hamlet,", msg.body)
 
+    # Helper for setting up various types of messages.
+    def setup_user_with_messages(self, user: UserProfile, admin: UserProfile) -> None:
+        self.send_personal_message(user, admin, content="private message 1")
+        self.send_personal_message(user, admin, content="private message 2")
+        self.send_stream_message(user, "Verona", topic_name="topic1", content="stream message 1")
+        self.send_stream_message(user, "Verona", topic_name="topic1", content="stream message 2")
+        self.send_stream_message(user, "Verona", topic_name="topic2")
+        self.send_stream_message(user, "Verona")
+
+        private_stream = self.make_stream("PrivateStream", invite_only=True)
+        self.subscribe(user, private_stream.name)
+        self.send_stream_message(
+            user, private_stream.name, topic_name="topic1", content="private stream message 1"
+        )
+        self.send_stream_message(
+            user, private_stream.name, topic_name="topic2", content="private stream message 2"
+        )
+
+    def test_deactivation_without_deleting_messages(self) -> None:
+        admin = self.example_user("othello")
+        do_change_user_role(admin, UserProfile.ROLE_REALM_ADMINISTRATOR, acting_user=None)
+        self.login("othello")
+        user = self.example_user("cordelia")
+        self.setup_user_with_messages(user, admin)
+
+        user_messages = list(get_user_sent_messages(user))
+        self.assert_length(user_messages, 8)
+
+        result = self.client_delete(
+            f"/json/users/{user.id}",
+            {
+                "message_delete_action": orjson.dumps(
+                    {
+                        "delete_public_stream_messages": False,
+                        "delete_private_stream_messages": False,
+                        "delete_direct_messages": False,
+                    }
+                ).decode(),
+            },
+        )
+        self.assert_json_success(result)
+        user_messages = list(get_user_sent_messages(user))
+        self.assert_length(user_messages, 8)  # No messages should be deleted
```

Comment:
Don't put comments on the end of a line; put them on the line before what they're describing.